sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/WritableColumnVector.java

```diff
@@ -586,7 +587,7 @@ public final int appendStruct(boolean isNull) {
     if (isNull) {
       appendNull();
       for (ColumnVector c: childColumns) {
-        if (c.type instanceof StructType) {
+        if (c.dataType() instanceof StructType) {
```

Comment:
nit: Which access type will we use for `ColumnVector.type`? `dataType()` or `type`?  
For example, `OnHeapColumnVector.reserveInternal()` uses `type` while this line uses `dataType()`.